### PR TITLE
[java-client] set collection to null when not required

### LINKED ISSUE #3585: [java][model] Collection initialized to empty collection even if not required
We are trying to use the same model between our client (using a model serialized with Jackson) and server (JaxRS jersey2).

If a collection is not required, the server model set its value to `null`.

https://github.com/OpenAPITools/openapi-generator/blob/7f36f2642209c6d7070e1d76209ba1928d816922/modules/openapi-generator/src/main/resources/JavaJaxRS/pojo.mustache#L24-L29

This is not the case on client side.

https://github.com/OpenAPITools/openapi-generator/blob/7f36f2642209c6d7070e1d76209ba1928d816922/modules/openapi-generator/src/main/resources/Java/pojo.mustache#L61

In my opinion this should be aligned.

### PR DESCRIPTION
**Pull request by jmini**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Fixes #3585

### CODE REVIEW

**Comment by jmini** [comment]
cc Java technical committee:
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala  (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01)  @karismann (2019/03) @Zomzog (2019/04)

File: pom.xml [hunk 1]
```diff
@@ -281,33 +281,6 @@
                     </execution>
                 </executions>
             </plugin>
-            <plugin>
```
Comment (wing328) [inline pom.xml]: @jmini Why this is removed? Is it causing issue?

**Comment by wing328** [comment]
@jmini I remember working on something similar and use `nullable` to determine the default value (null or empty list)

**Comment by wing328** [comment]
Related discussion/issue: https://github.com/OpenAPITools/openapi-generator/issues/1861

File: pom.xml (hunk 1 above)
Comment (jmini) [reply pom.xml]: I had problem locally yes, and I did not look into it in details yet.

It was not my intention to change it as part of this PR (my plan was to discuss it separately). I will revert this part for this PR.

**Comment by jmini** [comment]
> I remember working on something similar and use nullable to determine the default value (null or empty list)

As I wrote in issue #3585 this PR is aligning `Java/pojo.mustache` with `JavaJaxRS/pojo.mustache`

I will contribute to the discussion in #3615

**Comment by wing328** [comment]
> As I wrote in issue #3585 this PR is aligning Java/pojo.mustache with JavaJaxRS/pojo.mustache

Yes, I read that before leaving my comment. I'm not against this PR. Just want to share I've worked on something similar before with the `nullable` attribute.

**Pull request merged**